### ISSUE TITLE
IO::Socket::Async fails when using Inline::Perl5

### ISSUE THREAD

**Issue by zarulzakuan** [niner/Inline-Perl5#126]
How to replicate the issue:
Note that any P5 module will output the same error. I'm using Data::Dumper for this example. I am aware P6 has native Data::Dump module.

```
use Data::Dumper:from<Perl5>;
constant $SOCKET_PORT = 7000;
constant $SOCKET_ADDR = '0.0.0.0';
my $socket = IO::Socket::Async.listen($SOCKET_ADDR, $SOCKET_PORT);
react {
    whenever $socket -> $conn {
        whenever $conn.Supply(:bin) {
                say "Received --> "~$_.decode('utf-8');
        }
    }
}
```
Output:

```
Could not find symbol '&Async'
  in block <unit> at test.pl line 4
```
My Perl6 version

```
This is Rakudo version 2018.11-11-g07d95bf built on MoarVM version 2018.11
implementing Perl 6.d.
```

**Comment by lizmat** [comment]
This can be golfed to:

    use Data::Dumper:from<Perl5>;
    IO::Socket::Async

My guess is that because `IO::Socket` is a standard Perl 5 package, available in Perl 5 without doing a `use`, the search for `IO::Socket::Async` happens in Perl 5 space, where it doesn't exist.  So we effectively have the Perl 5 `IO::Socket` shadowing the Perl 6 one.

I don't see a simple solution to this short term.  I do see a workaround:

    constant IOSA = IO::Socket::Async;   # make sure we have access to Perl 6 version
    use Data::Dumper:from<Perl5>;
    constant $SOCKET_PORT = 7000;
    constant $SOCKET_ADDR = '0.0.0.0';
    my $socket = IOSA.listen($SOCKET_ADDR, $SOCKET_PORT);

Longer term, I guess we could be thinking about a generic `:from<>` syntax, so you could say:

    my $socket = IO::Socket::Async:from<Perl6>.listen($SOCKET_ADDR, $SOCKET_PORT);

**Comment by zarulzakuan** [comment]
@lizmat Thank you Liz! Guess I have to abandon my Go learning now :-)

**Comment by niner** [comment]
This is fixed by commit 00ee3d965fde2975ef758125e9d1ff6803d77517 which is part of Inline::Perl5 0.38. We now only create wrappers for the module requested and everything in the module's name space (e.g. use A:from<Perl5> will wrap A and A::B but not B).
